modules/lang-painless/src/main/java/org/elasticsearch/painless/phase/DefaultUserTreeToIRTreePhase.java

```diff
@@ -1235,16 +1403,40 @@ public void visitSymbol(ESymbol userSymbolNode, ScriptScope scriptScope) {
 
         if (scriptScope.hasDecoration(userSymbolNode, StaticType.class)) {
             Class<?> staticType = scriptScope.getDecoration(userSymbolNode, StaticType.class).getStaticType();
-            StaticNode irStaticNode = new StaticNode();
-            irStaticNode.setLocation(userSymbolNode.getLocation());
-            irStaticNode.setExpressionType(staticType);
-            irExpressionNode = irStaticNode;
+            StaticNode staticNode = new StaticNode();
+            staticNode.setLocation(userSymbolNode.getLocation());
+            staticNode.setExpressionType(staticType);
+            irExpressionNode = staticNode;
         } else if (scriptScope.hasDecoration(userSymbolNode, ValueType.class)) {
-            VariableNode irVariableNode = new VariableNode();
-            irVariableNode.setLocation(userSymbolNode.getLocation());
-            irVariableNode.setExpressionType(scriptScope.getDecoration(userSymbolNode, ValueType.class).getValueType());
-            irVariableNode.setName(userSymbolNode.getSymbol());
-            irExpressionNode = irVariableNode;
+            boolean read = scriptScope.getCondition(userSymbolNode, Read.class);
```

Comment:
Dot, symbol, brace must handle loads and/or stores.

Reply:
Done.